save-frontend/src/main/kotlin/com/saveourtool/save/frontend/utils/CComponent.kt

```diff
@@ -0,0 +1,61 @@
+@file:Suppress("FILE_NAME_INCORRECT")
+
+package com.saveourtool.save.frontend.utils
+
+import kotlinext.js.assign
+import react.*
+import react.Fragment
+
+import kotlinx.js.jso
+
+/**
+ * Base class that inherit [Component] class in order to provide [ChildrenBuilder] API in class components.
+ */
+@Suppress("CLASS_NAME_INCORRECT")
+abstract class CComponent<P : Props, S : State> : Component<P, S> {
+    constructor() : super() {
+        state = jso { init() }
+    }
+    constructor(props: P) : super(props) {
+        state = jso { init(props) }
+    }
+
+    @Suppress(
+        "WRONG_OVERLOADING_FUNCTION_ARGUMENTS",
+        "EMPTY_BLOCK_STRUCTURE_ERROR",
+        "MISSING_KDOC_CLASS_ELEMENTS",
+        "MISSING_KDOC_ON_FUNCTION",
+    )
+    open fun S.init() {}
```

Comment:
We didn't ever use similar method of `RComponent`, I think we should either remove it or rewrite state initializations from `init` to `override fun init`